app/actions/websocket/index.ts

```diff
@@ -64,90 +68,119 @@ export async function handleClose(serverUrl: string, lastDisconnect: number) {
     });
 }
 
-function doFirstConnect(serverUrl: string) {
-    updateAllUsersSinceLastDisconnect(serverUrl);
-}
-
 async function doReconnect(serverUrl: string) {
-    const database = DatabaseManager.serverDatabases[serverUrl];
-    if (!database) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
         return;
     }
 
-    const system = await queryCommonSystemValues(database.database);
-    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database.database);
-
-    // TODO consider fetch only and batch all the results.
-    await fetchMe(serverUrl);
-    await fetchMyPreferences(serverUrl);
-    const {config} = await fetchConfigAndLicense(serverUrl);
-    const {memberships: teamMemberships, error: teamMembershipsError} = await fetchMyTeams(serverUrl);
-    const {currentChannelId, currentUserId, currentTeamId, license} = system;
-    const currentTeamMembership = teamMemberships?.find((tm) => tm.team_id === currentTeamId && tm.delete_at === 0);
-
-    let channelMemberships: ChannelMembership[] | undefined;
-    if (currentTeamMembership) {
-        const {memberships, channels, error} = await fetchMyChannelsForTeam(serverUrl, currentTeamMembership.team_id, true, lastDisconnectedAt);
-        if (error) {
-            DeviceEventEmitter.emit(Events.TEAM_LOAD_ERROR, serverUrl, error);
-            return;
-        }
-        const currentUser = await queryCurrentUser(database.database);
-        const preferences = currentUser ? (await currentUser.preferences.fetch()) : [];
-        const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], system.config, license);
-        const directChannels = channels?.filter((c) => c.type === General.DM_CHANNEL || c.type === General.GM_CHANNEL);
+    const {database} = operator;
+    const tabletDevice = await isTablet();
+    const system = await queryCommonSystemValues(database);
+    const lastDisconnectedAt = await queryWebSocketLastDisconnected(database);
+
+    resetWebSocketLastDisconnected(operator);
+    let {config, license} = await fetchConfigAndLicense(serverUrl);
+    if (!config) {
```

Comment:
In what instance we may not receive neither of these?

Reply:
`fetchConfigAndLicense` return a promise of type ConfigAndLicenseRequest  and its definition is
```typescript
type ConfigAndLicenseRequest = {
    config?: ClientConfig;
    license?: ClientLicense;
    error?: unknown;
}
```

so this is just about being the safest possible in case that request does not include either config or license

Reply:
If it is about safety, shouldn't we fail if we don't get the config and license? Like... there has been an error connecting to the server, so probably the next requests will also fail. Using the "latest license and config I have" does not seem the safest thing to do.

But maybe I am missing something around here.

Reply:
Can you be sure that the reason for the failure is a massive failure? With that logic we should execute requests in series everywhere and cancel everything after the first one failed.

Basically in this case, I prefer to have the wrong assumption that the cinfig or license have not changed and use what is on disk at this moment

Reply:
> "latest license and config I have" does not seem the safest thing to do.

If we fail everything the app will continue to behave with the latest config and license I have anyway